airflow/jobs/triggerer_job_runner.py

```diff
@@ -751,3 +757,28 @@ def get_trigger_by_classpath(self, classpath: str) -> type[BaseTrigger]:
         if classpath not in self.trigger_cache:
             self.trigger_cache[classpath] = import_string(classpath)
         return self.trigger_cache[classpath]
+
+    @staticmethod
+    def add_trigger_cancel_reasons(
+        triggerer_id, cancel_trigger_ids: set[int]
+    ) -> list[tuple[int, TriggerTerminationReason]]:
+        """
+        Add trigger cancel reasons to give consumer more context.
+
+        Currently, we only distinguish between reassigned and other reasons.
+        """
+
+        def add_reason(
+            ids: set[int], reason: TriggerTerminationReason
+        ) -> list[tuple[int, TriggerTerminationReason]]:
+            return [(trigger_id, reason) for trigger_id in ids]
+
+        # find out reassigned triggers
+        reassigned_trigger_ids = set(Trigger.filter_out_reassigned_triggers(triggerer_id, cancel_trigger_ids))
+
+        other_reasons_trigger_ids = cancel_trigger_ids - reassigned_trigger_ids
+
+        trigger_id_reasons = add_reason(reassigned_trigger_ids, TriggerTerminationReason.REASSIGNED)
+        trigger_id_reasons.extend(add_reason(other_reasons_trigger_ids, TriggerTerminationReason.OTHER))
+
+        return trigger_id_reasons
```

Comment:
```suggestion
        reassigned_pairs = (
            (trigger_id, TriggerTerminationReason.REASSIGNED)
            for trigger_id in reassigned_trigger_ids
        )
        other_reason_pairs = (
            (trigger_id, TriggerTerminationReason.OTHER)
            for trigger_id in other_reasons_trigger_ids
        )
        return [*reassigned_pairs, *other_reason_pairs]
```

Same as your code but easier to read IMO. The additional `add_reason` function doesn’t make things easier.

Reply:
I agree. I initially thought the reasons might be expanded here, but since there are currently only two types, let's adopt your suggestion